runtime/tests/checker/arrays_dictionaries_test.go

```diff
@@ -317,6 +317,91 @@ func TestCheckDictionaryValues(t *testing.T) {
 	)
 }
 
+func TestCheckDictionaryEqual(t *testing.T) {
+	t.Parallel()
+
+	testValid := func(name, code string) {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			_, err := ParseAndCheck(t, code)
+			require.NoError(t, err)
+		})
+	}
+
+	assertInvalid := func(name, code string) {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			_, err := ParseAndCheck(t, code)
+			errs := RequireCheckerErrors(t, err, 1)
+			assert.IsType(t, &sema.InvalidBinaryOperandsError{}, errs[0])
+		})
+	}
+
+	for _, opStr := range []string{"==", "!="} {
+		testValid("self_dict_equality", fmt.Sprintf(`
+			fun test(): Bool {
+				let d = {"abc": 1, "def": 2}
+				return d %s d
+			}`, opStr))
```

Comment:
Minor: Could you please chop down the argument lists? That would make the code a bit more readable.
For example here:

```suggestion
		testValid(
			"self_dict_equality", 
			fmt.Sprintf(
				`
					fun test(): Bool {
						let d = {"abc": 1, "def": 2}
						return d %s d
					}
				`,
				opStr,
			),
		)
```

Reply:
Done for all the tests.